assembl/static2/js/app/components/common/richTextEditor/index.jsx

```diff
@@ -0,0 +1,191 @@
+
+  static defaultProps = {
+    handleInputFocus: null,
+    maxLength: 0,
+    toolbarPosition: 'top'
+  };
+
+  constructor(props: RichTextEditorProps): void {
+    super(props);
+    this.state = {
+      editorState: EditorState.createEmpty(),
+      editorHasFocus: false
+    };
+
+    if (props.rawContentState) {
+      this.state.editorState = EditorState.createWithContent(convertFromRaw(props.rawContentState));
+    }
+  }
+
+  componentDidMount() {
+    this.editor.focus();
+  }
+
+  componentWillReceiveProps(nextProps: RichTextEditorProps): void {
+    // we want to reset the editor state only if rawContentState is null (i.e. in case the form has been reset)
+    if (!nextProps.rawContentState) {
+      this.setState({
+        editorState: EditorState.createEmpty()
+      });
+    }
+  }
+
+  onBlur = () => {
+    const rawContentState = convertToRaw(this.state.editorState.getCurrentContent());
+    this.setState({
+      editorHasFocus: false
+    });
+    this.props.updateContentState(rawContentState);
+  };
+
+  onChange = (newEditorState: EditorState): void => {
+    this.setState({
+      editorState: newEditorState
+    });
+  };
+
+  handleEditorFocus = (): Function => {
+    const { handleInputFocus } = this.props;
+    this.setState({
+      editorHasFocus: true
+    });
+
+    return handleInputFocus;
+  };
+
+  getToolbarButtons(): Array<ButtonConfigType> {
+    const bold = {
+      id: 'bold',
+      icon: 'text-bold',
+      label: I18n.t('common.editor.bold'),
+      type: 'style',
+      style: 'BOLD'
+    };
+    const italic = {
+      id: 'italic',
+      icon: 'text-italics',
+      label: I18n.t('common.editor.italic'),
+      type: 'style',
+      style: 'ITALIC'
+    };
+    const bullets = {
+      id: 'bullets',
+      icon: 'text-bullets',
+      label: I18n.t('common.editor.bulletList'),
+      type: 'block-type',
+      style: 'unordered-list-item'
+    };
+    const buttons = [bold, italic, bullets];
+    return buttons;
+  }
+
+  getCharCount(editorState: EditorState): number {
+    // this code is "borrowed" from the draft-js counter plugin
+    const decodeUnicode = (str) => {
+      return punycode.ucs2.decode(str);
+    }; // func to handle unicode characters
+    const plainText = editorState.getCurrentContent().getPlainText('');
+    const regex = /(?:\r\n|\r|\n)/g; // new line, carriage return, line feed
+    const cleanString = plainText.replace(regex, '').trim(); // replace above characters w/ nothing
+    return decodeUnicode(cleanString).length;
+  }
+
+  shouldHidePlaceholder(): boolean {
+    // don't display placeholder if user changes the block type (to bullet list) before to type anything
+    const contentState = this.state.editorState.getCurrentContent();
+    if (!contentState.hasText()) {
+      if (contentState.getBlockMap().first().getType() !== 'unstyled') {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  focusEditor = (): void => {
```

Comment:
Ah right, the toolbar doesn't work at all without the setTimeout. I'll add a comment in the code.